jpsonic-main/src/main/java/com/tesshu/jpsonic/dao/MediaFileDao.java

```diff
@@ -865,56 +870,97 @@ select count(id)
     }
 
-    public void updateAlbumSort(SortCandidate candidate) {
+    public void updateAlbumSort(SortCandidate cand) {
         template.update("""
                 update media_file
                 set album_reading = ?, album_sort = ?
                 where present and id = ?
-                """, candidate.getReading(), candidate.getSort(), candidate.getId());
+                """, cand.getReading(), cand.getSort(), cand.getTargetId());
     }
 
-    public void updateArtistSort(SortCandidate candidate) {
-        if (candidate.getField() == CandidateField.ARTIST) {
+    public void updateArtistSort(ArtistSortCandidate cand) {
+        if (cand.getTargetField() == TargetField.ARTIST && cand.getTargetType() == MediaType.DIRECTORY) {
+            template.update("""
+                    update media_file
+                    set artist_reading = ?, artist_sort = ?, music_index = ?
+                    where id = ?
+                    """, cand.getReading(), cand.getSort(), cand.getMusicIndex(), cand.getTargetId());
+        } else if (cand.getTargetField() == TargetField.ARTIST) {
             template.update("""
                     update media_file
                     set artist_reading = ?, artist_sort = ?
                     where id = ?
-                    """, candidate.getReading(), candidate.getSort(), candidate.getId());
-        } else if (candidate.getField() == CandidateField.ALBUM_ARTIST) {
+                    """, cand.getReading(), cand.getSort(), cand.getTargetId());
+        } else if (cand.getTargetField() == TargetField.ALBUM_ARTIST) {
             template.update("""
                     update media_file
                     set album_artist_reading = ?, album_artist_sort = ?
                     where id = ?
-                    """, candidate.getReading(), candidate.getSort(), candidate.getId());
-        } else if (candidate.getField() == CandidateField.COMPOSER) {
+                    """, cand.getReading(), cand.getSort(), cand.getTargetId());
+        } else if (cand.getTargetField() == TargetField.COMPOSER) {
             template.update("""
                     update media_file
                     set composer_sort = ?
                     where id = ?
-                    """, candidate.getSort(), candidate.getId());
+                    """, cand.getSort(), cand.getTargetId());
         }
     }
 
+    public void updateArtistSort(List<ArtistSortCandidate> cands) {
+        final String updates = "update media_file ";
+        final StringBuilder cols = new StringBuilder(130);
+        cols.append("set ");
+        final String cond = "where id = ?";
+
+        List<Object> args = new ArrayList<>();
+        cands.forEach(cand -> {
+            switch (cand.getTargetField()) {
+            case ARTIST -> {
+                cols.append("artist_reading=?, artist_sort=?, ");
+                args.add(cand.getReading());
+                args.add(cand.getSort());
+            }
```

Comment:
Case is missing.
`cand.getTargetField() == TargetField.ARTIST && cand.getTargetType() == MediaType.DIRECTORY`